### DISCUSSION TITLE
Why does Mantine rerender on hover (Menu, Select, ...)

### DISCUSSION THREAD

**Discussion by b2m9** [mantinedev/mantine#2806] [Q&A]
Just toying around with Mantine and I noticed that it re-renders the entire parent component when hovering over Menu items or Select items. In source you can see that it updates the currently hovered list item index. This by itself shouldn't trigger a rerender, though. What am I missing here? Can I opt out of this behavior?

**Comment by rtivital** [comment]
When option is hovered it update state in Select component to keep track of current hovered item, it cannot be changed

**Reply by b2m9** [reply]
What is the use case for updating component state based on hover? If it's indeed necessary, why can't the index of the currently hovered item be stored as a ref? Re-rendering based on hover just seems a bit excessive to me, but happy to learn why it is necessary.

**Reply by rtivital** [reply]
When state updates in root component, all components that use this state rerender

**Reply by b2m9** [reply]
> When state updates in root component, all components that use this state rerender

Yes, that's clear. But _why_ do you store the index of the hovered list item as `state`? What's the use case here? Re-rendering a select box on mouse enter/leave for each item seems excessive. Why is it necessary to re-render? It cannot just be to highlight the current item, right? If it's not necessary, then why not store the hovered list item index as a `ref`?

**Reply by rtivital** [reply]
It is required for keyboard events support, up/down arrows change hovered index
